requirements/usage/usage-scenarios.tex

```diff
@@ -0,0 +1,27 @@
+\subsubsection{Usage scenarios}
+\begin{enumerate}
+  \item High priority task.
+  \begin{itemize}
+    \item Bob has to do some urgent work.
+	\item Bob inputs the task and data. He also sets the priority option to high.
+	\item System checks if the task can be directly executed somewhere.
+	If that is the case, the task is executed and the scenario ends.
+	\item System tries to find the running task with lowest priority.

```

Comment:
I am not sure we'll always want to interrupt the task with lowest priority. @JohannesGaessler What do you think?

Reply:
Why exactly are you not sure?
Because not all machines are equally fast and you would want to run high-priority tasks on fast machines?

Reply:
Yes, for example this. Also it could be that most lower priority tasks run on low-end machines but we need a high-end computer for the new task.

Reply:
I think determining which lower-priority task to stop is not a simple problem.
To solve this problem I think we need to work out how we're going to schedule tasks in general.

Reply:
Agreed. Maybe we can just say something generic here and refer to the algorithm we're going to have to develop.

Reply:
Something like
**System tries to find the most suitable running task, that can be paused and replaced by Bob's**

Reply:
This is missing that Bob's task should only be run if there are tasks with lower priority currently running.
Also so far we didn't consider that tasks require varying amounts of resources.
How about this:

System checks if there are running tasks with a priority that is lower than Bob's task.
System checks how many resources can be freed by pausing the lower-priority tasks.
If enough resources can be freed the system automatically pauses one or more lower-priority tasks and runs Bob's task instead.

Reply:
Looks good to me. I will wait for @ammen99 to comment and then I will change it.

Reply:
@JohannesGaessler I really believe we should just stick to the `find the most suitable task(s) that can be paused`. Later on, we may want to add deadlines or user priorities (so a user automatically has high/low priority/whatever).

Reply:
That being said, I'm not adamant about the wording here. If you both think it is better to use alternatives, do so.

Reply:
I think this isn't just about wording.
The issue at hand is how detailed our descriptions of scenarios should be.

Reply:
I'd say this depends on how many scenarios we will have. If we have multiple scenarios that touch this point (rescheduling existing tasks) then it will definitely make more sense to have it all in one place, referenced by the scenarios.

Anyway, we will need to have a detailed description of the algorithm in the functional requirements. So we don't need to go into details here, but still we shouldn't have conflicting information in both places.

Reply:
http://www.agilemodeling.com/artifacts/usageScenario.htm
Here they write:
`"Usage scenarios can be very detailed, indicating exactly how someone works with the user interface, or reasonably high-level describing the critical business actions but not the indicating how they're performed."`
So i think both ways are fine.

Reply:
How about this:

System checks if enough resources can be freed for Bob's task by pausing one or more appropriate running tasks.